app/controllers/api/v2/base_controller.rb

```diff
@@ -89,7 +89,8 @@ def metadata_page
       end
 
       def metadata_per_page
-        @per_page ||= params[:per_page].present? ? params[:per_page].to_i : Setting[:entries_per_page]
+        @per_page ||= Setting[:entries_per_page] if params[:per_page].empty?
+        @per_page ||= params[:per_page] == 'all' ? metadata_total : params[:per_page].to_i
```

Comment:
I'm wondering if we should allow `per_page=all` for any controller or make it opt in. I can imagine it's an easy way to take down a server if you request all hosts at once on a large installation. On the other hand, we already see people specify `per_page=999999999` to disable pagination.

Reply:
also (a bit unrelated) hammer host list I think sends a big number to list all hosts I think

I'd say for consistency reasons, we should enable it everywhere, it just shouldn't be the default value, people who are switching this settings should know what they are doing. We may want to mention that in the setting description

Reply:
999999999? amateurs :)

https://github.com/theforeman/foreman-ansible-modules/blob/fdf454497dbcb50372ee3adfa8f1e41587b4dcca/plugins/module_utils/foreman_helper.py#L439:
```python
params['per_page'] = 2 << 31
```

Reply:
I'd guess it'll help us there as well xD